DEPLOYMENT: Mehrere Integrationsdeployments mit demselben Codestand ermöglichen

### Aktuelles Verhalten

Wenn mehrmals hintereinander in demselben Composition Repo die "Update Integration" Action ausgeführt wird, ohne dass der Code im Core und Wagons sich geändert hat, schlägt der Workflow fehl:
https://github.com/hitobito/ose_composition_cevi/actions/runs/6972009650/job/18973319197

```
nothing to commit, working tree clean
Command exited with a non-zero exit-code
Exitstatus: pid 2450 exit 1
 -> git commit -m 'Update integration to 1.31.6-7' ⛌
Error: Process completed with exit code 1.
```

### Gewünschtes Verhalten

Zum Debugging / zur Entwicklung der Composition Repos / zum neu Setzen von Environment Variabeln und Secrets kann derselbe Codestand auch mehrmals hintereinander auf Integration (und optimalerweise auch auf Produktion) ausgerollt werden.

### Schritte zu reproduzieren

Zwei Mal hintereinander den Update Integration Workflow ankicken

### Wann ist der Fehler aufgetreten?

23.11.2023

--- a/lib/release/highlevel.rb
+++ b/lib/release/highlevel.rb
@@ -70,6 +70,12 @@ module Release
                     failure: 'No changes in translations'
     end
 
+    def changes_to_be_committed?
+      execute_check 'test $(git diff --name-only --staged | wc -l) -gt 0',
+                    success: 'Staged changes found',
+                    failure: 'No staged changes found'
+    end
+
     def update_version(file:, to: @version)
       notify "writing version to #{file}"
       case file
@@ -79,7 +85,7 @@ module Release
         execute %(sed -i "s/VERSION\s*=\s*'[0-9.]*'/VERSION = '#{to}'/" #{file})
       end
       add file
-      commit 'Bump Version for Release'
+      commit 'Bump Version for Release' if changes_to_be_committed?
     end
 
     def update_changelog(file: 'CHANGELOG.md', to: @version)
@@ -109,7 +115,7 @@ module Release
     def record_submodule_state(with: @message)
       add 'hitobito'
       add 'hitobito_*'
-      commit with
+      commit with if changes_to_be_committed?
       submodule_status
     end
 

--- a/lib/release/lowlevel.rb
+++ b/lib/release/lowlevel.rb
@@ -38,7 +38,7 @@ module Release
     end
 
     def submodules(action)
-      notify 'executing action in all submodule'
+      notify "executing '#{action}' in all submodules"
       execute "git submodule foreach '#{action}'"
     end
 

